pulp_smash/tests/puppet/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,650 @@
+        response = requests.get(_PUPPET_QUERY_URL)
+        response.raise_for_status()
+        cls.original_puppets.append(response.content)
+
+        # Create two repos, start upload request, upload module to upload dir
+        cls.attrs_iter = tuple((
+            create_repository(cls.cfg, _gen_puppet_repo_body())
+            for _ in range(2)
+        ))
+        cls.bodies['upload_malloc'] = _start_content_upload(
+            cls.cfg,
+            cls.responses['upload_malloc']
+        )
+        cls.bodies['upload'] = _upload_file(
+            cls.cfg,
+            cls.bodies['upload_malloc']['_href'],
+            cls.original_puppets[0],
+            cls.responses['upload'],
+        )
+
+        # Import uploaded puppet modules into first repo. Poll.
+        # Delete upload request.
+        cls.bodies['import'] = _import_puppet_to_repo(
+            cls.cfg,
+            cls.bodies['upload_malloc']['upload_id'],
+            cls.attrs_iter[0]['_href'],
+            cls.responses['import'],
+        )
+        for call_report in cls.bodies['import']:
+            poll_spawned_tasks(cls.cfg, call_report)
+        cls.bodies['upload_free'] = delete(
+            cls.cfg,
+            cls.bodies['upload_malloc']['_href'],
+            cls.responses['upload_free'],
+        )
+
+        # Copy content from first repo to second. Poll until done.
+        cls.bodies['copy'] = _copy_repo(
+            cls.cfg,
+            cls.attrs_iter[0]['id'],
+            cls.attrs_iter[1]['_href'],
+            cls.responses['copy'],
+        )
+        for call_report in cls.bodies['copy']:
+            poll_spawned_tasks(cls.cfg, call_report)
+
+        # Add a distributor to the repositories, then publish repositories
+        # to the distributors. Poll the publish request.
+        # First repository
+        cls.bodies['distribute'] = tuple((
+            _add_puppet_distributor(
+                cls.cfg,
+                cls.attrs_iter[i]['_href'],
+                cls.responses['distribute']
+            ) for i in range(2)
+        ))
+        cls.bodies['publish'] = tuple((
+            _publish_repo(
+                cls.cfg,
+                cls.attrs_iter[i]['_href'],
+                cls.bodies['distribute'][i]['id'],
+                cls.responses['publish'],
+            ) for i in range(2)
+        ))
+        for i in range(2):
+            for call_report in cls.bodies['publish'][i]:
+                poll_spawned_tasks(cls.cfg, call_report)
+
+        # Query modules on server. Data integrity is tested in tests below.
+        hrefs = []
+        hrefs.append(''.join((
+            '/v3/releases?module=',
+            _PUPPET_QUERY_PACKAGE['name'],
+            '/',
+            _PUPPET_QUERY_PACKAGE['author']
+        )))
+        hrefs.append(''.join((
+            '/v3/releases?module=',
+            _PUPPET_QUERY_PACKAGE['name'],
+            '/',
+            _PUPPET_QUERY_PACKAGE['author']
+        )))
```

Comment:
The two elements of `hrefs` are identical, so I see no reason to create an iterable. Just create a single variable. In addition, using `format` is more readable.

```python
href = '/v3/releases?module={}/{}'.format(
    _PUPPET_QUERY_PACKAGE['name'],
    _PUPPET_QUERY_PACKAGE['author'],
)
```

That done, you can drop the `i` variable from the loop below.